src/providers/EtherspotContextProvider.tsx

```diff
@@ -378,6 +379,38 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getRatesByTokenAddresses = useCallback(async (chainId: number, tokenAddresses: string[]) => {
+    let tokens  = tokenAddresses.filter((address) => address !== null)
+    if(!sdk || !tokens.length) return null
+    try {
+      const rates: RateData = await sdk.fetchExchangeRates({tokens, chainId});
+      if(!rates.errored && rates.items.length > 0) {
+        let addressByChain : Record<string, number> = {}
+        rates.items.forEach((rate) => {
+          addressByChain[rate.address] = rate.usd
+        })
+        return addressByChain
```

Comment:
Should we change the var name `addressByChain`? It doesn't really represent what's being returned

Reply:
nah I guess it should be something like `rateByAddress`